File: examples/08_ffi/java/complex_plugin/build-package.java
import dev.daedalus.plugin.PackageBuilder;
import ffi.showcase.ShowcasePlugin;

final class BuildPackage {
  public static void main(String[] args) throws Exception {
    PackageBuilder.fromAnnotatedPlugin(ShowcasePlugin.class)
        .classesDir("build/classes/java/main")
        .jar("build/libs/ffi-showcase.jar")
        .nativeLibrary("build/native/libffi_showcase_jni.so")
        .write("plugin.json");
  }
}

File: examples/08_ffi/cpp/complex_plugin/build-package.cpp
#include <daedalus.hpp>

#include "src/showcase.cpp"

int main() {
  daedalus::PackageBuilder::from_plugin("ffi_showcase")
      .shared_library("build/libffi_showcase.so")
      .source_file("src/showcase.cpp")
      .write("plugin.json");
}
